test/uiauto/base.js

```diff
@@ -72,34 +70,29 @@ async function newInstruments (bootstrapFile) {
 }
 
 async function init (bootstrapFile, opts) {
-  let deferred = Promise.pending();
-  let proxy = new CommandProxy(opts);
-  let instruments;
-  proxy.start(
-    // first connection
-    function (err) {
+  return new Promise(async (resolve, reject) => {
+    let proxy = new UIAutoClient(opts);
+    let instruments;
+
+    // only want to resolve the promise when we actually connect
+    await proxy.start((err) => {
       instruments.launchHandler(err);
-      if (err) return deferred.reject(err);
-      deferred.resolve({proxy: proxy, instruments: instruments});
-    },
-    // regular cb
-    async (err) => {
-      if (err) return deferred.reject(err);
-      instruments = await newInstruments(bootstrapFile);
-      instruments.start(null, function () {
-        proxy.safeShutdown(_.noop);
-        deferred.reject('Unexpected shutdown of instruments');
-      });
-    }
-  );
-  return deferred.promise;
+      if (err) return reject(err);
+      resolve({proxy, instruments});
+    });
+    instruments = await newInstruments(bootstrapFile);
```

Comment:
I suppose we might need to update this once @sebv's appium-instruments rewrite is done